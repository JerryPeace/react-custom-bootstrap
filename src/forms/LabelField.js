import PropTypes from 'prop-types';
import React from 'react';
import {Basic} from './Basic';
import Label from './Label';
import FormGroup from './FormGroup';
import FormControls from './FormControls';
import classNames from 'classnames';
import LayoutHelper from './LayoutHelper';

class LabelField extends Basic {
    static propTypes = {
        labelText: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
        value: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
        standalone: PropTypes.bool
    };

    static defaultProps = {
        labelAlign: 'left',
        standalone: true,
        bsClass: 'label-field',
        labelClassName: 'col-xs-2',
        wrapperClassName: 'col-xs-10'
    };

    renderInput() {
        const {value, className} = this.props;

        return (
            <div className={className}>
                <span>
                    {(typeof value === 'string') ? value.replace(/  /gm, ' \u00a0') : value}
                </span>
            </div>
        );
    }

    render() {
        return (
            <LayoutHelper FieldBase={this.renderInput()} {...this.props} />
        );
    }
}

export default LabelField;
