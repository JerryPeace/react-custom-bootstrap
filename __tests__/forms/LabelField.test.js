import React from 'react';
import {mount} from 'enzyme';
import LabelField from 'forms/LabelField';

describe('LabelField', () => {
    let labelFeild;

    test('with default settings', () => {
        labelFeild = mount(
            <LabelField
                labelText='label text'
                value='value test'
            >
                children
            </LabelField>
        );
        expect(labelFeild).toMatchSnapshot();
    });

    test('with multiple whitespace which should be translated into (0x20, 0xa0) in bytes', () => {
        labelFeild = mount(
            <LabelField
                labelText='label text'
                value='multiple       whitespace and   second    one'
            >
                children
            </LabelField>
        );
        expect(labelFeild).toMatchSnapshot();
    });
});
