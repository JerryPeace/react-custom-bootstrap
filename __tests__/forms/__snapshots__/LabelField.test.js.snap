// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelField with default settings 1`] = `
<LabelField
  bsClass="label-field"
  labelAlign="left"
  labelClassName="col-xs-2"
  labelText="label text"
  standalone={true}
  value="value test"
  wrapperClassName="col-xs-10"
>
  <LayoutHelper
    FieldBase={
      <div
        className={undefined}
      >
        <span>
          value test
        </span>
      </div>
    }
    bsClass="label-field"
    labelAlign="left"
    labelClassName="col-xs-2"
    labelText="label text"
    singleCheckboxRadio={false}
    standalone={true}
    value="value test"
    wrapperClassName="col-xs-10"
  >
    <div
      className=""
    >
      <Label
        FieldBase={
          <div
            className={undefined}
          >
            <span>
              value test
            </span>
          </div>
        }
        className="col-xs-2"
        inline={false}
        labelAlign="left"
        text="label text"
        value="value test"
        wrapperClassName="col-xs-10"
      >
        <label
          className="col-xs-2"
        >
          label text
        </label>
      </Label>
      <div
        className="form-control-static"
      >
        <span>
          value test
        </span>
      </div>
      children
    </div>
  </LayoutHelper>
</LabelField>
`;

exports[`LabelField with multiple whitespace which should be translated into (0x20, 0xa0) in bytes 1`] = `
<LabelField
  bsClass="label-field"
  labelAlign="left"
  labelClassName="col-xs-2"
  labelText="label text"
  standalone={true}
  value="multiple       whitespace and   second    one"
  wrapperClassName="col-xs-10"
>
  <LayoutHelper
    FieldBase={
      <div
        className={undefined}
      >
        <span>
          multiple       whitespace and   second    one
        </span>
      </div>
    }
    bsClass="label-field"
    labelAlign="left"
    labelClassName="col-xs-2"
    labelText="label text"
    singleCheckboxRadio={false}
    standalone={true}
    value="multiple       whitespace and   second    one"
    wrapperClassName="col-xs-10"
  >
    <div
      className=""
    >
      <Label
        FieldBase={
          <div
            className={undefined}
          >
            <span>
              multiple       whitespace and   second    one
            </span>
          </div>
        }
        className="col-xs-2"
        inline={false}
        labelAlign="left"
        text="label text"
        value="multiple       whitespace and   second    one"
        wrapperClassName="col-xs-10"
      >
        <label
          className="col-xs-2"
        >
          label text
        </label>
      </Label>
      <div
        className="form-control-static"
      >
        <span>
          multiple       whitespace and   second    one
        </span>
      </div>
      children
    </div>
  </LayoutHelper>
</LabelField>
`;
